torch/csrc/jit/script/compiler.cpp

```diff
@@ -563,6 +564,55 @@ struct to_ir {
     ConstantPooling(to_clean);
   }
 
+void runInline(Block* block) {
+
+  Node* cur = block->nodes().front();
+  Node* end = block->return_node();
+
+  while (cur != end)
```

Comment:
I did consider iterators, but since we need to keep track of several iteration points, it looked rather clunky.